### PR TITLE
Feature/non embedded landmark capabilities

### PR DESCRIPTION
**Pull request by simonmitchell**
## Description

- Adds ability to provide a timestamp to `downloadFullBundle` function on `ContentController`
- Adds function to `ContentController` to remove all bundles (delta, original, temporary)

## Motivation and Context

This is required to allow work for non-bundled storm apps (GDPC) to adhere to landmark publishes! They will remove and re-download all content each time the app is launched.

## How Has This Been Tested?

Used locally in GDPC First Aid
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ThunderCloud/ContentController.swift [hunk 1]
```diff
@@ -1694,6 +1696,30 @@ public class ContentController: NSObject {
...
+        
+        if let _deltaBundle = deltaDirectory {
+            removeBundle(in: _deltaBundle)
+        }
+        
+        if let _tempDirectory = temporaryUpdateDirectory {
+            removeBundle(in: _tempDirectory)
+        }
+        
+        // Remove intial bundle timestamp as we've now cleared out all evidence of any bundles!
```
Comment (BenCShutt) [inline ThunderCloud/ContentController.swift:1718]: Minor typo

File: ThunderCloud/ContentController.swift [hunk 2]
```diff
@@ -366,37 +366,39 @@ public class ContentController: NSObject {
...
-        // Remove intial bundle timestamp as we've now cleared out all evidence of any bundles!
-        initialBundleTimestamp = nil
-        
-        configureBaseURL()
+        guard var urlComponents = URLComponents(string: "\(baseString)/\(version)/apps/\(appId)/bundle") else {
+            fatalError("Failed to create url to download a storm bundle: one of Base URL, Api Version or AppID are invalid url parts")
+        }
+            
+        guard let bundleDirectory = bundleDirectory else {
+            return
```
Comment (BenCShutt) [inline ThunderCloud/ContentController.swift:387]: What does it mean if we get here?

File: ThunderCloud/ContentController.swift [hunk 3]
```diff
@@ -366,37 +366,39 @@ public class ContentController: NSObject {
...
-            if let _fullBundleURL = URL(string: "\(baseString)/\(version)/apps/\(appId)/bundle"), let _destinationURL = bundleDirectory {
-                // We set the initial bundle timestamp here because this bundle will now act as the app's initial bundle!
-                downloadPackage(fromURL: _fullBundleURL, destinationDirectory: _destinationURL, setAsInitialBundle: true, progressHandler: progressHandler)
-            }
+        guard let bundleUrl = urlComponents.url else {
+            fatalError("Failed to create url to download a storm bundle: one of Base URL, Api Version or AppID are invalid url parts")
         }
         
+        // We set the initial bundle timestamp here because this bundle will now act as the app's initial bundle!
+        downloadPackage(fromURL: bundleUrl, destinationDirectory: bundleDirectory, setAsInitialBundle: true, progressHandler: progressHandler)
```
Comment (BenCShutt) [inline ThunderCloud/ContentController.swift:401]: If this fails we've deleted all existing bundles?

File: ThunderCloud/ContentController.swift [hunk 4]
```diff
@@ -366,37 +366,39 @@ public class ContentController: NSObject {
     
     /// Downloads a full storm content bundle, this will clear all directories and will also mark the downloaded bundle as the 'initial' bundle timestamp
     /// so that we can avoid downloading any post-landmark publishes from content-available notifications!
-    /// - Parameter progressHandler: A closure which will be called with the progress of the bundle download
-    public func downloadFullBundle(with progressHandler: ContentUpdateProgressHandler?) {
+    /// - Parameter buildTimestamp: The timestamp of the build, used to make sure we don't bypass any landmark publishes
```
Comment (BenCShutt) [inline ThunderCloud/ContentController.swift:369]: Might be worth documenting what these seconds represent, unix time?

File: ThunderCloud/ContentController.swift (hunk 2 above)
Comment (simonmitchell) [reply ThunderCloud/ContentController.swift:387]: Yeah I was unsure here whether to `fatalError` or not... it basically means the directory which stores the default content failed to be created:

```
//Identify folders for bundle
            if let _bundlePath = NSSearchPathForDirectoriesInDomains(.applicationSupportDirectory, .userDomainMask, true).last {
                
                let _bundleDirectory = URL(fileURLWithPath: _bundlePath, isDirectory: true).appendingPathComponent("StormBundle")
                
                bundleDirectory = _bundleDirectory
                
                //Create application support directory
                do {
                    try FileManager.default.createDirectory(atPath: _bundleDirectory.path, withIntermediateDirectories: true, attributes: nil)
                } catch {
                    baymax_log("Failed to create bundle directory at \(_bundleDirectory.absoluteString)", subsystem: Logger.stormSubsystem, category: ContentController.logCategory, type: .fault)
                    os_log("Failed to create bundle directory at %@", log: contentControllerLog, type: .fault, _bundleDirectory.absoluteString)
                }
            }
```

I didn't feel like here was a valid place to `fatalError` as everywhere else that uses this property just fails silently

File: ThunderCloud/ContentController.swift (hunk 3 above)
Comment (simonmitchell) [reply ThunderCloud/ContentController.swift:401]: You are correct... luckily the only app that uses this are the GDPC apps which handle all failures really gracefully including on a new app launch ☺️

File: ThunderCloud/ContentController.swift (hunk 4 above)
Comment (simonmitchell) [reply ThunderCloud/ContentController.swift:369]: Yep, good shout!

File: ThunderCloud/ContentController.swift (hunk 2 above)
Comment (BenCShutt) [reply ThunderCloud/ContentController.swift:387]: I guess in this case will could try and re-download the bundle? Possibly by going down the root of first app launch

**Pull request merged**